src/javaharness/java/arcs/crdt/CRDTCollection.java

```diff
@@ -0,0 +1,174 @@
+package arcs.crdt;
+
+import java.lang.Math;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+// Classes and interfaces copied from src/runtime/crdt/crdt-collection.ts
+interface Referenceable {
+  String getId();
+}
+
+class VersionedValue<T> {
+  T value;
+  VersionMap version;
+  VersionedValue(T value, VersionMap version) {
+    this.value = value;
+    this.version = version;
+  }
+}
+
+class CollectionData<T extends Referenceable> extends CRDTData {
+   Map<String, VersionedValue<T>> values = new HashMap<>();
+}
+
+enum CollectionOpTypes { ADD, REMOVE }
+
+class CollectionOperation<T> implements CRDTOperation {
+  CollectionOpTypes type;
+  Optional<T> added;
+  Optional<T> removed;
+  String actor;
+  VersionMap clock;
+}
+
+class RawCollection<T> extends HashSet<T> implements CRDTConsumerType {
+  RawCollection() {
+    super();
```

Comment:
I think this is implicit in Java: https://stackoverflow.com/a/2054040

Reply:
removed, thanks.